app/src/main/java/org/jellyfin/androidtv/util/profile/LibVlcProfile.kt

```diff
@@ -0,0 +1,146 @@
+
+import org.jellyfin.androidtv.constant.CodecTypes
+import org.jellyfin.androidtv.constant.ContainerTypes
+import org.jellyfin.androidtv.util.DeviceUtils
+import org.jellyfin.androidtv.util.Utils
+import org.jellyfin.androidtv.util.profile.ProfileHelper.deviceHevcCodecProfile
+import org.jellyfin.androidtv.util.profile.ProfileHelper.subtitleProfile
+import org.jellyfin.apiclient.model.dlna.CodecProfile
+import org.jellyfin.apiclient.model.dlna.CodecType
+import org.jellyfin.apiclient.model.dlna.DirectPlayProfile
+import org.jellyfin.apiclient.model.dlna.DlnaProfileType
+import org.jellyfin.apiclient.model.dlna.ProfileCondition
+import org.jellyfin.apiclient.model.dlna.ProfileConditionType
+import org.jellyfin.apiclient.model.dlna.ProfileConditionValue
+import org.jellyfin.apiclient.model.dlna.SubtitleDeliveryMethod
+
+class LibVlcProfile(
+	isLiveTV: Boolean = false
+) : BaseProfile() {
+	init {
+		name = "AndroidTV-libVLC"
+
+		directPlayProfiles = arrayOf(
+			// Video direct play
+			DirectPlayProfile().apply {
+				type = DlnaProfileType.Video
+
+				container = arrayOf(
+					ContainerTypes.M4V,
+					ContainerTypes._3GP,
+					ContainerTypes.TS,
+					ContainerTypes.MPEGTS,
+					ContainerTypes.MOV,
+					ContainerTypes.XVID,
+					ContainerTypes.VOB,
+					ContainerTypes.MKV,
+					ContainerTypes.WMV,
+					ContainerTypes.ASF,
+					ContainerTypes.OGM,
+					ContainerTypes.OGV,
+					ContainerTypes.M2V,
+					ContainerTypes.MPG,
+					ContainerTypes.MPEG,
+					ContainerTypes.MP4,
+					ContainerTypes.WEBM,
+					ContainerTypes.WTV
+				).joinToString(",")
+
+				audioCodec = listOfNotNull(
+					CodecTypes.AAC,
+					CodecTypes.MP3,
+					CodecTypes.MP2,
+					CodecTypes.AC3,
+					CodecTypes.WMA,
+					CodecTypes.WMAV2,
+					CodecTypes.DCA,
+					CodecTypes.DTS,
+					CodecTypes.PCM,
+					CodecTypes.PCM_S16LE,
+					CodecTypes.PCM_S24LE,
+					CodecTypes.OPUS,
+					CodecTypes.FLAC,
+					CodecTypes.TRUEHD,
+					if (!Utils.downMixAudio() && isLiveTV) CodecTypes.AAC_LATM else null
+				).joinToString(",")
+			},
+			// Audio direct play
+			DirectPlayProfile().apply {
+				type = DlnaProfileType.Audio
+
+				container = arrayOf(
+					CodecTypes.FLAC,
+					CodecTypes.AAC,
+					CodecTypes.MP3,
+					CodecTypes.MPA,
+					CodecTypes.WAV,
+					CodecTypes.WMA,
+					CodecTypes.MP2,
+					ContainerTypes.OGG,
+					ContainerTypes.OGA,
+					ContainerTypes.WEBMA,
+					CodecTypes.APE
+				).joinToString(",")
+			},
+			// Photo direct play
+			DirectPlayProfile().apply {
+				type = DlnaProfileType.Photo
+				container = "jpg,jpeg,png,gif"
+			}
+		)
+
+		codecProfiles = arrayOf(
+			// HEVC profile
+			deviceHevcCodecProfile,
+			// H264 profile
+			CodecProfile().apply {
+				type = CodecType.Video
+				codec = CodecTypes.H264
+				conditions = arrayOf(
+					ProfileCondition(
+						ProfileConditionType.EqualsAny,
+						ProfileConditionValue.VideoProfile,
+						"high|main|baseline|constrained baseline"
```

Comment:
`arrayOf().joinToString()`

Reply:
Is it worth using `joinToString()` here given the space in there? I feel like that would be harder to read.

Reply:
I'd say it would be easier to read where the split happens, because "constrained baseline" is a single value, which is not obvious at first glance.